File: python/src/cours.py
from .interfaces import ICours

class Cours(ICours):
    def __init__(self, nom: str, professeur: str):
        self._nom = nom
        self._professeur = professeur

    @property
    def nom(self) -> str:
        return self._nom

    @property
    def professeur(self) -> str:
        return self._professeur

    def __str__(self):
        return f"Cours: {self.nom} | Professeur: {self.professeur}"

File: java/src/Cours.java
public class Cours {
    private String nom;
    private String professeur;

    public Cours(String nom, String professeur) {
        this.nom = nom;
        this.professeur = professeur;
    }

    public String getNom() {
        return nom;
    }

    public String getProfesseur() {
        return professeur;
    }

    @Override
    public String toString() {
        return "Cours: " + nom + " | Professeur: " + professeur;
    }
}
